### DISCUSSION TITLE
Import a Camino Stejskal Tanner scheme file in DIPY

### DISCUSSION THREAD

**Discussion by nadluru** [dipy/dipy#2433] [3. Community Lounge]
Dear DIPY community,

Thank you for building such an amazing ecosystem around diffusion imaging! I was wondering if there is a command/tool in DIPY to convert the full Camino's [Stejskal Tanner scheme file](http://camino.cs.ucl.ac.uk/index.php?n=Docs.SchemeFiles) to DIPY's gradient table which can then be used to fit diffusion models. Could you please let me know of a way to do it if currently there is no direct API for this?

Thank you so much for your help and advice.
Sincerely,
Nagesh

**Comment by arokem** [comment]
Hi @nadluru -- thanks for your question. We don't currently have an API for that. But judging from the page you sent, I don't think that it would be too hard to write one. It looks like this file format is a pure text format that numpy should be able to read with a `loadtxt` and then you'd have to do some. I would encourage you to try to implement this and make a PR. I'd be happy to review this after you give it a go. This should probably go into this file: https://github.com/dipy/dipy/blob/master/dipy/io/gradients.py

**Reply by nadluru** [reply]
Hi @arokem,

Thanks so much for the prompt reply! I see. I will try to work on this and seek help if needed. I think I need to work on an additional file that creates the "gradient_table" object as well. I believe I have to work on this file (https://github.com/dipy/dipy/blob/master/dipy/core/gradients.py) too.

Sincerely,
Nagesh
